tla-types/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestToEtcExprRows.scala

```diff
@@ -0,0 +1,84 @@
+package at.forsyte.apalache.tla.typecheck.etc
+
+import at.forsyte.apalache.io.annotations.store.{createAnnotationStore, AnnotationStore}
+import at.forsyte.apalache.io.typecheck.parser.{DefaultType1Parser, Type1Parser}
+import at.forsyte.apalache.tla.lir.UntypedPredefs._
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.convenience.tla
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatestplus.junit.JUnitRunner
+
+/**
+ * Unit tests for translating TLA+ expressions to EtcExpr using rows.
+ *
+ * @author
+ *   Igor Konnov
+ */
+@RunWith(classOf[JUnitRunner])
+class TestToEtcExprRows extends AnyFunSuite with BeforeAndAfterEach with ToEtcExprBase {
+  private var parser: Type1Parser = _
+  private var annotationStore: AnnotationStore = _
+  private var gen: ToEtcExpr = _
+
+  override protected def beforeEach(): Unit = {
+    parser = DefaultType1Parser
+    annotationStore = createAnnotationStore()
+    // a new instance of the translator, as it gives unique names to the variables
+    gen = new ToEtcExpr(annotationStore, ConstSubstitution.empty, new TypeVarPool(), useRows = true)
+  }
+
+  test("record set constructor") {
+    val funOperType = parser("(Set(a), Set(b)) => Set({ x: a, y: b })")
+    val expected = mkConstAppByName(funOperType, "S", "T")
+    assert(expected == gen(tla.recSet(tla.str("x"), tla.name("S"), tla.str("y"), tla.name("T"))))
+  }
+
+  test("invalid field string in record set construction") {
+    val invalid = "invalidName"
```

Comment:
I don't get why this field name is invalid, can you help me understand?

Reply:
Good point. I have added a comment. The test is checking that only `ValEx(TlaStr(_))` can be passed, not `NameEx(_)`.